File: TaskManagerAPI/TaskManagerMvc/ViewModels/SignUpViewModel.cs
using System;
using TaskManagerMvc.Models;

namespace TaskManagerMvc.ViewModels
{
	public class SignUpViewModel
	{
		public PersonFullName PersonName { get; set; }
		public string Email { get; set; }
		public string Mobile { get; set; }
		public string DateOfBirth { get; set; }
		public string Password { get; set; }
		public string Gender { get; set; }
		public int CountryID { get; set; }
		public bool RecieveNewsLetters { get; set; }
		public List<Skill> Skills { get; set; }

		public SignUpViewModel()
		{
		}
	}

	public class PersonFullName
	{
		public string FirstName { get; set; }
		public string LastName { get; set; }
	}
}

File: Angular/TaskManager/src/app/sign-up-view-model.ts
export class SignUpViewModel {
    personName:any;
    email:string|any;
    mobile:string|any;
    dateOfBirth:|string|any;
    password:string|any;
    gender:string|any;
    countryID:number|any;
    recieveNewsLetters:boolean|any;
    skills:any;

    /**
     *
     */
    constructor(
        personName:any = null,
        email:string|any = null,
        mobile:string|any = null,
        dateOfBirth:|string|any = null,
        password:string|any = null,
        gender:string|any = null,
        countryID:number|any = null,
        recieveNewsLetters:boolean|any = null,
        skills:any = null
    ) {
       
        this.personName = personName;
        this.email = email;
        this.mobile = mobile;
        this.dateOfBirth = dateOfBirth;
        this.password = password;
        this.gender = gender;
        this.countryID = countryID;
        this.recieveNewsLetters = recieveNewsLetters;
        this.skills = skills;
    }
}
